File: Source/Easy_Longest_Common_Prefix.cpp
#include <string>
#include <vector>
#include <iostream>

std::string Longest_Common_Prefix(std::vector<std::string>& strs)
{
	unsigned int i = strs[0].length();

	for(unsigned int s = 0; s < strs.size() - 1; s++)
	{
		for(unsigned int c = 0; c < i; c++)
		{
			if(c >= strs[s].length()) {i = strs[s].length(); break;}
			if(c >= strs[s+1].length()) {i = strs[s+1].length(); break;}
			if(strs[s][c] != strs[s+1][c]) i = c;
		}
	}

	return strs[0].substr(0, i);
}

std::vector<std::string> strs1({"a"});
std::vector<std::string> strs2({"",""});
std::vector<std::string> strs3({"ab","a"});
std::vector<std::string> strs4({"flower","flow","flight"});

int main()
{
	std::cout << Longest_Common_Prefix(strs1) << '\n';
	std::cout << Longest_Common_Prefix(strs2) << '\n';
	std::cout << Longest_Common_Prefix(strs3) << '\n';
	std::cout << Longest_Common_Prefix(strs4) << '\n';
}

File: Source/Easy_Longest_Common_Prefix.cs
using System;

public class Solution
{
	public String Longest_Common_Prefix(String[] strs)
	{
		int i = strs[0].Length;

		for(int s = 0; s < strs.Length - 1; s++)
		{
			for(int c = 0; c < i; c++)
			{
				if(c >= strs[s].Length) {i = strs[s].Length; break;}
				if(c >= strs[s+1].Length) {i = strs[s+1].Length; break;}
				if(strs[s][c] != strs[s+1][c]) i = c;
			}
		}

		return strs[0].Substring(0, i);
	}
}

public class LeetCode
{
	public static void Main()
	{
		String[] strs1 = new string[] {"a"};
		String[] strs2 = new string[] {"",""};
		String[] strs3 = new string[] {"ab","a"};
		String[] strs4 = new string[] {"flower","flow","flight"};

		Solution solution = new Solution();
		Console.WriteLine(solution.Longest_Common_Prefix(strs1));
		Console.WriteLine(solution.Longest_Common_Prefix(strs2));
		Console.WriteLine(solution.Longest_Common_Prefix(strs3));
		Console.WriteLine(solution.Longest_Common_Prefix(strs4));
	}
}